instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy/patches/client.rb

```diff
@@ -103,19 +103,14 @@ def detect_unmatched_pairs(obfuscated)
           end
 
           def database_span_name(sql)
-            # Setting span name to the SQL query without obfuscation would
-            # result in PII + cardinality issues.
-            # First attempt to infer the statement type then fallback to
-            # current Otel approach {database.component_name}.{database_instance_name}
-            # https://github.com/open-telemetry/opentelemetry-python/blob/39fa078312e6f41c403aa8cad1868264011f7546/ext/opentelemetry-ext-dbapi/tests/test_dbapi_integration.py#L53
-            # This creates span names like mysql.default, mysql.replica, postgresql.staging etc.
-
-            statement_type = extract_statement_type(sql)
-
-            return statement_type unless statement_type.nil?
-
-            # fallback
-            'mysql'
+            # TODO: When https://github.com/github/trilogy/pull/48/commits/6e2377ae2d737962a7a1a46b8f63438868034c7b is released,
+            # we will have access to the database name, so we can support :span_name configs of :db_name and :db_operation_and_name.
+            case config[:span_name]
```

Comment:
Is there a way we can evaluate this once? Instead of switching everytime?

Reply:
wdyt about just keeping `:db_operation_and_name` and the behavior of the `mysql2` instrumentation being consistent

```ruby
            when :db_operation_and_name
              op = OpenTelemetry::Instrumentation::Mysql2.attributes['db.operation']
              name = database_name
              if op && name
                "#{op} #{name}"
              elsif op
                op
              elsif name
                name
              end
```

In this case we can let `database_name` be a no-op (or just add a code comment with the TODO), document the configuration limitation in instrumentation readme. I'd like to avoid a place where we add, and then immediately deprecate, the `:db_operation` configuration option.

Reply:
Makes sense! The `database_name` should be available in the next release, yes, just not sure when that will be 😄

Reply:
> but that seems like a fair amount of added complexity to avoid a fairly simple case statement. What do you think?

My instinct is to replace conditional logic with polymorphism but not required. 

In our case at GH we will always use `statement_type` so the case statement will always hit the first conditional so I suspect it won't have much of an impact for us to add a `===` call for each SQL span. If I see anything come up in our profiles I will report back to you all after we test this release.

Reply:
@ericmustin I have not kept a close eye on Trilogy releases but if `database_name` becomes available that would be pretty great.

Reply:
Daniel Colson from Github's Ruby Archeology team has kindly offered to cut a new release of Trilogy tomorrow, so I'll get the proposed changes in tmrw.

Reply:
@composerinteralia 🙇🏼